lib/api/routes/iterations.rb

```diff
@@ -11,17 +11,29 @@ class Iterations < Core
         require_key
 
         if current_user.guest?
```

Comment:
No, it only checks if you supplied the `:key` param. It doesn't do anything with authentication. You still need to explicitly test if a user is guest. 

Seems a little bit strange to me at a first sight, but I have no idea if this is by design or not. Might be worth to moving the code a bit and have the `requires_key` method also check if the key is valid. From a quick search it seems that `requires_key` is always used together with `guest?`. We could easily get rid of some repetition. 

Let me know what you think. I am happy to open new PR for this if it is something you would like.